alternator/server.cc

```diff
@@ -80,6 +80,10 @@ class api_handler : public handler_base {
                      resf.get();
                  } catch (api_error &ae) {
                      ret = ae;
+                 } catch (rjson::invalid_parameter& ivp) {
+                     ret = api_error("InvalidParameterValue", ivp.what());
+                 } catch (rjson::missing_parameter& mp) {
+                     ret = api_error("MissingParameter", mp.what());
                  } catch (rjson::error & re) {
```

Comment:
From what I could tell, all existing places that expect a ValidationException iff something is missing already does so manually. I based this on just checking some places for what is expected as error if a parameter is missing, though I will wholly admit it was not exhaustive. Agree needs better testing than done here. 
That is why its an RFC.

Reply:
What I meant was that you went into a lot of trouble (not really related to the Streams feature, probably doesn't need to be in the same series by the way) to add this "InvalidParameterValue" and "MissingParameter" error types, but I never actually saw them in DynamoDB runs. Before we start to use these error types for some purpose, I want to have test cases which demonstrates that DynamoDB uses these error types for this purpose.

Reply:
Tests == good. Agree wholly. 
The error  selection is based on the API spec. Of course, it could be lying.

Reply:
There is conflicting documentation, so you'll **have** to check.
One page https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/CommonErrors.html indeed lists "MissingParameter" and "InvalidParameterValue". However, the heading of this page says "This section lists the errors common to the API actions of all AWS services.".
The more DynamoDB-centric page https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Programming.Errors.html has a different list, doesn't list those two and instead lists only "ValidationException", explaining it as "This error can occur for several reasons, such as a required parameter that is missing, a value that is out of range, or mismatched data types.".

This is why all Alternator tests can run against DynamoDB, not just against Alternator - we usually write them before writing the Scylla code, and can verify how DynamoDB works in successful and erroneous cases, so we can do the same thing in Scylla instead of just guessing how DynamoDB handles cases where the documentation is unclear or ambigous.